app/categories/README.md

```diff
@@ -0,0 +1,129 @@
+# Category Traversal System
+
+The category traversal system is responsible for handling routing the user through the category diagnosis questions.
+
+## Overview
+
+The category traversal system is designed to
+- Handle multi-step questionnaires with branching logic
+- Pre-calculate all possible navigation paths
+- Support both internal and external redirects
+- Provide a mapping of the users answers to each question
+- Provide back links remembering the users previous answer
+
+Whilst keeping the service stateless and without requiring session data to be stored by the user.
+
+This system implements a [Mealy Finite State Machine](https://en.wikipedia.org/wiki/Mealy_machine), where the next state
+is only a function of the current state and an input.
+
+If business requirements require the next state to be a function of the users previous answers then this can be achieved
+by splitting the path and having two parallel routes depending on the users previous answer.
+
+## How to add a question
+To add new question
+
+1. Create a new question form class inheriting from `QuestionForm`:
+
+    Each question form must define:
+   - `category`: The category of law the question belongs to
+   - `title`: The question text populating the page title
+   - `question`: A WTForms form field containing the actual question
+   - `routing_logic`: A dictionary mapping answer choices to their next steps
+
+Example:
+```python
+class MyQuestionForm(QuestionForm):
+    title = "What type of help do you need?"
+    
+    routing_logic = {
+        "yes": NextQuestionForm,  # Question forms should only be initalised when handling the users request.
+        "no": CheckRedirect(destination=CheckDestination.FALA)
+    }
+    
+    question = RadioField(
+        title,
+        widget=CategoryRadioInput(),  # Uses our override class to support setting custom CSS on the label title
+        validators=[InputRequired(message="Validation failed message")],
+        choices=[
+            ("yes", "Yes"),
+            ("no", "No"),
+        ],
+    )
+```
+
+2. Add your new question form to the routing logic of an existing question:
+```python
+class ExistingQuestionForm(QuestionForm):
+    routing_logic = {
+        "existing-choice": ExistingNextStep,
+        "new-choice": NewQuestionForm  # Add your new form here
```

Comment:
It can do, I tend to not use the `MyObject` syntax. However, if there is value in it I am more than happy to.

In this case I am discussing adding a new question form.

Reply:
Good catch! Yes, you're right!